pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -964,6 +967,31 @@ type MSTeamsConfig struct {
 	HTTPConfig *HTTPConfig `json:"httpConfig,omitempty"`
 }
 
+// MSTeamsV2Config configures notifications via Microsoft Teams using the new message format with adaptive cards as required by flows
+// See https://prometheus.io/docs/alerting/latest/configuration/#msteamsv2_config
+// It requires Alertmanager >= 0.28.0.
+type MSTeamsV2Config struct {
+	// Whether to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// MSTeams incoming webhook URL. It is mutually exclusive with `webhookUrlFile`.
+	// Either `webhookUrl` or `webhookUrlFile` are required.
+	// +optional
+	WebhookURL *v1.SecretKeySelector `json:"webhookUrl,omitempty"`
+	// File to read the MSTeams webhook URL from.
+	// +optional
+	WebhookURLFile *string `json:"webhookUrlFile,omitempty"`
```

Comment:
Though this is not very consistent with other receivers, let's support only secret key reference for now. Files are a bit more problematic because they assume that the user knows where the url file lives in the Alertmanager container.
```suggestion
```